src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java

```diff
@@ -152,6 +154,7 @@ public MqttClient connect(Handler<AsyncResult<MqttConnAckMessage>> connectHandle
         this.connection = new MqttClientConnection(this, soi, options);
 
         soi.messageHandler(msg -> connection.handleMessage(msg));
+        soi.closeHandler((v) -> handleClosed());
```

Comment:
sorry to be pedantic ... one parameter no need for parenthesis :-) After that you can squash and I'll merge